core/src/main/scala/sudoku/sudoku.scala

```diff
@@ -122,23 +122,33 @@ case class Sudoku(grid: Vector[Vector[Cell]]) {
 
 object Sudoku {
 
-  def parse(s: String): Option[Sudoku] = {
-    sequence(s
-      .toVector
-      .map {
-        case '.' => Some(Undetermined((1 to 9).toList))
-        case w => Try(w.toString.toInt).toOption.map(Fixed)
-      }).map(grid => Sudoku(grid
-      .grouped(9)
-      .toVector))
+  def validateSize(s: String): Either[Throwable, Unit] = if (s.length == 9 * 9) Right(()) else Left(new IllegalArgumentException("Grid string must be 81 characters long"))
+
+  def validateCell(c: Char): Either[Throwable, Int] = {
```

Comment:
I think it is better to create an error ADT as you can easily check you get the correct error in your test, e.g.
```
sealed trait ParsingError
case class InvalidGridSize(n: Int) extends ParsingError
case class InvalidCell(c: Char) extends Parsing Error
```